src/harness/util.py

```diff
@@ -25,3 +29,92 @@ def decorated_testcase(*args, **kwargs):
     testcase(*args, **kwargs)
 
   return decorated_testcase
+
+
+def _getRandomLatLongInPolygon(polygon):
+  
+  try:
+    geojson_ppa = geojson.loads(json.dumps(polygon['features'][0]['geometry']))
+    ppa_polygon = shape(geojson_ppa)
+    min_lng, min_lat, max_lng, max_lat = ppa_polygon.bounds
+    lng = random.uniform(min_lng, max_lng)
+    lng_line = LineString([(lng, min_lat), (lng, max_lat)])
+    lng_line_intercept_min, lng_line_intercept_max = \
+      lng_line.intersection(ppa_polygon).xy[1].tolist()
+    lat = random.uniform(lng_line_intercept_min, lng_line_intercept_max)
+    if Point([lng, lat]).within(ppa_polygon):
+      return lat, lng
+    else:
+      return _getRandomLatLongInPolygon(polygon)
+  except NotImplementedError:
+    # If there's no intercept again call the function
+    return _getRandomLatLongInPolygon(polygon)
+
+
+def _changeDateInPpaAndPal(record, prev_date, next_date):
+  for key, value in record.iteritems():
+    if type(record[key]) is dict:
+      _changeDateInPpaAndPal(record[key], prev_date, next_date)
+    record[key] = (next_date if 'Expiration' in key else
+                  prev_date if 'Date' in key else record[key])
+  return record
+
+
+def makePpaAndPalRecordsConsistent(ppa_record, pal_records,
+                                   ppa_fips_code, ppa_census_year,
+                                   low_frequency, high_frequency, device):
+  """Make PPA, PAL and Device object consistent with the inputs and position 
+  the device in the PPA Polygon at some random location
+
+    Args:
+      ppa_record: (dictionary) An object containing PPA Record.
+      pal_records: (list) A list of PAL Records which has to be
+        associated with the PPA.
+      ppa_fips_code: (number) The FIPS Code of the Census Tract where 
+      the PPA belongs.
+      ppa_census_year: (number) The census year in which the census 
+      tract was defined.
+      low_frequency: (number) The Primary Low Frequency for PAL
+      high_frequency: (number) The Primary High Frequency for PAL
+      device: (dictionary) An object containing device which has to be
+      moved into the PPA Polygon.
+    Returns:
+      A tuple consisting of PPA Record, PAL Record List and Device with 
+      modified Latitude and Longitude
+  """
+
+  previous_year_date = datetime.now().replace(year=datetime.now().year - 1)
+  next_year_date = datetime.now().replace(year=datetime.now().year + 1)
+
+  for pal_record in pal_records:
+    # Change the FIPS Code and Registration Date-Year in Pal Id
+    pal_record['palId'] = '/'.join([str(ppa_fips_code) if index == 2 else
```

Comment:
This feels overly complicated. Why not just use:

```
fcc_channel_id = "1"   # you will need to use this later to fill licenseFrequencyChannelId
pal_record['palId'] = "/".join(["pal", #same date code, fcc_channel_id])
```

The current implementation looks like it might break in interesting ways if the PAL ID in the input is malformed.